typebackend/typingviews.py

```diff
@@ -51,17 +35,28 @@ class Paradetails(APIView):
     permission_classes=[IsAuthenticated]
    
     def get(self,request):
-        
-        global previous_para_number
-        total_no_of_paragraph=Paragraph.objects.count()
-        para_position=randint(0,total_no_of_paragraph-1)
+        global previous_typed
+        typed_para=[]
+
+        userlog=PractiseLog.objects.filter(user=request.user)
 
-        if(previous_para_number==para_position):
-            para_position=generateNewParaNumber(previous_para_number,total_no_of_paragraph)
+        for data in userlog:
+            typed_para_id=data.para.id-1
+            typed_para.append(typed_para_id)
 
+        total_no_of_paragraph=Paragraph.objects.count()-1
+        para_position=randint(0,total_no_of_paragraph)
+
+        if total_no_of_paragraph in typed_para:
+            if previous_typed==para_position:
+                para_position=generateNewParaNumber([para_position],total_no_of_paragraph)
+        else:
+            if para_position in typed_para:
+                para_position=generateNewParaNumber(typed_para,total_no_of_paragraph)
+            
```

Comment:
All this logic can be simplified to two lines using just the right django queries.
Read about `exclude` and [in](https://docs.djangoproject.com/en/3.0/ref/models/querysets/#in).